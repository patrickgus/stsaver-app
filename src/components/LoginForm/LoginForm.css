.LoginForm {
  display: flex;
  flex-direction: column;
}

.LoginForm div {
  margin-bottom: 15px;
}

.LoginForm__header {
  text-align: center;
}

.LoginForm__buttons {
  display: flex;
  justify-content: space-around;
}

.LoginForm__buttons button {
  width: 100px;
}
