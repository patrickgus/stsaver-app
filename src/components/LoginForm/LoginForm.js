import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import "./LoginForm.css";

class LoginForm extends Component {
  handleSubmit = () => {
    this.props.history.push("/activity");
  };

  handleDemoClick = () => {
    this.props.history.push("/activity");
  };

  render() {
    return (
      <form className="LoginForm" onSubmit={this.handleSubmit}>
        <div className="LoginForm__header">
          <h2>Login Here</h2>
        </div>
        <div className="user_name">
          <label htmlFor="LoginForm__username">Username</label>
          <input type="text" name="username" id="LoginForm__username" />
        </div>
        <div className="password">
          <label htmlFor="LoginForm__password">Password</label>
          <input name="password" type="password" id="LoginForm__password" />
        </div>
        <button type="submit">Login</button>
        <button className="LoginForm__demo" onClick={this.handleDemoClick}>
          Demo
        </button>
      </form>
    );
  }
}

export default withRouter(LoginForm);
