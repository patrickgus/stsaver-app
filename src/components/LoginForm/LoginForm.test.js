import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import LoginForm from "./LoginForm";

describe("LoginForm component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <LoginForm />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the form as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <LoginForm />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
