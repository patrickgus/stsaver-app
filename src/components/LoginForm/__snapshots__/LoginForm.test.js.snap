// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders the form as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    className="LoginForm__header"
  >
    <h2>
      Login Here
    </h2>
  </div>
  <div
    role="alert"
  />
  <div
    className="LoginForm__form-section"
  >
    <label
      htmlFor="username"
    >
      Username
    </label>
    <input
      className="Input "
      id="username"
      name="username"
      type="text"
    />
  </div>
  <div
    className="LoginForm__form-section"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="Input "
      id="password"
      name="password"
      type="password"
    />
  </div>
  <div
    className="LoginForm__buttons"
  >
    <button
      className="Button "
      type="submit"
    >
      Login
    </button>
    <button
      className="Button LoginForm__demo"
      onClick={[Function]}
    >
      Demo
    </button>
  </div>
</form>
`;
